### PR TITLE
1deg_jra55do_ryf: change case name to "access-om3"

### PR DESCRIPTION
**Pull request by dougiesquire**
This PR changes the case name from "GMOM_JRA" to "access-om3". See https://github.com/COSIMA/access-om3/issues/106. The restart manifests accidentally added in https://github.com/COSIMA/MOM6-CICE6/pull/18 are also removed here.

The plan is to add a check to the ACCESS-OM3 Payu driver to check that the naming in the MOM `diag_table` is consistent with the case name set in `nuopc.runconfig`.

Once this PR is merged, I'll cherry pick into `1deg_jra55do_iaf`.

### CODE REVIEW

File: manifests/input.yaml [hunk 1]
```diff
@@ -4,8 +4,8 @@ version: 1.0
 work/input/JRA55do-ESMFmesh.nc:
   fullpath: /g/data/ik11/inputs/access-om3/0.x.0/share/meshes/JRA55do-ESMFmesh.nc
   hashes:
-    binhash: 5195a7517bd0905add0f97ceecc0503f
-    md5: 154215afc3b742e3da9ac77e4eac6813
+    binhash: 1147443c3fe237a480af1b4700159e21
```
Comment (anton-seaice) [inline manifests/input.yaml:7]: Why has this file changed?

File: manifests/restart.yaml
Comment (anton-seaice) [inline manifests/restart.yaml:1]: Can you please put this in a seperate commit to the rest of the change? (And then do a "Rebase and merge" when you merge it).

File: manifests/input.yaml (hunk 1 above)
Comment (dougiesquire) [reply manifests/input.yaml:7]: It changed when the mesh generation script got moved to the `om3-scripts` repo (all that changed is the metadata about how the file was generated - the contents are the same)

Comment (anton-seaice) [reply manifests/input.yaml:7]: Maybe that needs its own commit too :)

File: manifests/restart.yaml
Comment (dougiesquire) [reply manifests/restart.yaml:1]: Done

**Review by anton-seaice — APPROVED** [review]
I ran this as written and with a new build from [access-om3](https://github.com/anton-seaice/access-om3/commit/4b4c1892c3444f67f7bd98045922c63f0953e5f8) and it works. All the history output files (except [mom_rho2](https://github.com/COSIMA/MOM6-CICE6/issues/40)) and restart files are there with the `access-om3` name. Its possible to write cmeps history output too.

**Pull request merged**